src/activities/ActivityUsageEntity.js

```diff
@@ -71,31 +71,105 @@ export class ActivityUsageEntity extends Entity {
 		return dueDate && dueDate.properties.date;
 	}
 
+	/**
+	 * @returns {bool} Whether or not you have the ability to edit the due date
+	 */
 	canEditDueDate() {
-		const dueDate = this._getDueDateSubEntity();
-		return (dueDate && dueDate.hasActionByName(Actions.activities.update))
+		return this._canEditDate( Classes.dates.dueDate )
+	}
+
+	/**
+	 * Updates the due date of the activity usage entity
+	 * @param {string} dateValue The due date to be set
+	 */
+	async setDueDate(dateValue) {
+		await this._setDate(dateValue, Classes.dates.startDate, 'dueDate');
+	}
+
+	/**
+	 * @returns {bool} The start date of the activity usage
+	 */
+	startDate() {
+		const startDate = this._getDateSubEntity(Classes.dates.startDate);
+		return startDate && startDate.properties.date;
+	}
+
+	/**
+	 * @returns {bool} Whether or not you have the ability to edit the start date
+	 */
+	canEditStartDate() {
+		return this._canEditDate( Classes.dates.startDate )
+	}
+
+	/**
+	 * Updates the start date of the activity usage entity
+	 * @param {string} dateValue The start date to be set
+	 */
+	async setStartDate(dateValue) {
+		await this._setDate(dateValue, Classes.dates.startDate, 'startDate');
+	}
+
+	/**
+	 * @returns {bool} The end date of the activity usage
+	 */
+	endDate() {
+		const endDate = this._getDateSubEntity(Classes.dates.endDate);
+		return endDate && endDate.properties.date;
+	}
+
+	/**
+	 * @returns {bool} Whether or not you have the ability to edit the end date
+	 */
+	canEditEndDate() {
+		return this._canEditDate( Classes.dates.endDate )
+	}
+
+	/**
+	 * Updates the end date of the activity usage entity
+	 * @param {string} dateValue The end date to be set
+	 */
+	async setEndDate(dateValue) {
+		await this._setDate( dateValue, Classes.dates.endDate, 'endDate');
+	}
+
+	_getDateSubEntity(dateClass) {
+		return this._entity
+			&& this._entity.getSubEntityByClass(dateClass);
+	}
+
+	/**
+	 * @param {string} dateClass The class for the date entity
+	 * @returns {bool} Whether or not you have the ability to edit a specific date 
+	 */
+	_canEditDate( dateClass ) {
+		const date = this._getDateSubEntity(dateClass);
+		return (date && date.hasActionByName(Actions.activities.update))
 			|| this._entity.hasActionByName(Actions.activities.startAddNew);
 	}
 
-	async setDueDate(dateString) {
-		if (!this.canEditDueDate) {
+	/**
+	 * Updates either start, end or due date of the activity usage entity
+	 * @param {string} dateValue The date value to be set
+	 * @param {string} dateClass The class for the date to update
+	 * @param {string} dateField The field of the date to update
+	 */
+	async _setDate(dateValue, dateClass, dateField) {
+		if (!this._canEditDate(dateClass)) {
 			return;
 		}
 
 		let action;
-		const dueDateEntity = this._getDueDateSubEntity();
-		if (dueDateEntity) {
-			action = dueDateEntity.getActionByName(Actions.activities.update);
+		const dateEntity = this._getDateSubEntity(dateClass);
```

Comment:
Still irks me to see fetching sub entities by class instead of rel. Surely the class ought to be "date" and the rel ought to be "due" 🤷‍♀

Reply:
Oh, the actions are different anyway, so one kind of date doesn't quack like a different kind of date 🤔

Reply:
Yeah, the field name for each date is the name of the date - you are correct that it probably could have just been `date`.

Reply:
well... the endpoint for updating each date is the same (the usage endpoint), so the field names have to be unique for each date.

Reply:
🤦‍♂ Right, totally correct. For some reason I thought we could have them on the same endpoint, and also the same field name, and... idk, magically know which one to update? Not sure why I thought that 😛